src/string.cr

```diff
@@ -2401,29 +2397,71 @@ class String
   #
   # "abcdef".compare("ABCDEF", case_insensitive: true) # => 0
   # "abcdef".compare("ABCDEG", case_insensitive: true) # => -1
+  #
+  # "heIIo".compare("heııo", case_insensitive: true, Unicode::CaseOptions::Turkic) # => 0
   # ```
-  def compare(other : String, case_insensitive = false)
-    return self <=> other unless case_insensitive
+  def compare(other : String, case_insensitive = false, options = Unicode::CaseOptions::None)
+    comparison = self <=> other
+    return comparison unless case_insensitive
 
-    reader1 = Char::Reader.new(self)
-    reader2 = Char::Reader.new(other)
-    ch1 = reader1.current_char
-    ch2 = reader2.current_char
+    if ascii_only? && other.ascii_only?
+      if self.size != other.size
+        others_first_char = other.at(0) do
+          return 1
+        end
+        if self.starts_with?(others_first_char)
+          if self.size > other.size
```

Comment:
Yes, I'm just copying `String#<=>`'s behavior here.
When both strings start with the same character, then the length of both is the decisive factor.
`"aAA" <=> "aA"` returns `1` because the first string is longer.
`"Hzz" <=> "Hzzz"` returns `-1` because the latter string is shorter.

Reply:
That's not correct. The strings are compared, char by char. If you end up with no chars in one of the strings then the other is "greater". Check the logic in the existing branch.

Reply:
Oh, just copying the logic of the non-ASCII branch worked perfectly.